src/main/java/seedu/duke/Storage.java

```diff
@@ -1,14 +1,196 @@
+            System.out.println("* Created new file for use *");
+        } else {
+            System.out.println("* Existing file detected *");
+
+        }
+        return file;
     }
 
-    public void writeToFile(TransactionList transactions) {
+    /**
+     * Initializes the duke.txt by checking its existence, then store the data values to the program.
+     *
+     * @return The TransactionList storing entries which would be used in the program.
+     * @throws StorageInputCorruptedException If there are errors due to corrupted duke.txt data.
+     * @throws StorageWriteErrorException     If the file could not be created or could not be written.
+     */
+    public TransactionList initializeFile() throws MoolahException {
+        try {
+            File file = checkIfFileExist();
+            Scanner input = new Scanner(file);
+            storeFileValuesLocally(storedTransactions, input);
+            System.out.println("* duke.txt loaded successfully! *");
+
+        } catch (MoolahException e) {
+            //Catch any parsing errors and throw the default StorageInputCorruptedException
+            throw new StorageInputCorruptedException();
+        } catch (IOException e) {
+            throw new StorageWriteErrorException();
+        }
+        return storedTransactions;
     }
+
+    /**
+     * Stores values from duke.txt to the program by parsing each line in the file.
+     *
+     * @param storedTransactions The TransactionList to hold the stored values from the file.
+     * @param input              The input from duke.txt to be processed.
+     * @throws MoolahException WHen there are parsing errors, due to corrupted data.
+     */
+
+    private void storeFileValuesLocally(TransactionList storedTransactions, Scanner input) throws MoolahException {
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern(DateFormats.DATE_STORAGE_OUTPUT_PATTERN.toString());
+
+        Command command = null;
+        while (input.hasNext()) {
+            String line = input.nextLine();
+            String[] splits = line.split(DELIMITER);
+            if (splits.length != 5) {
+                throw new StorageInputCorruptedException();
+            }
+
+            String type = splits[0];
+            String category = splits[1];
+            String amountString = splits[2];
+            //Date has been formatted in duke.txt and must be synthesized into the correct string format before parsing.
+            try {
+                LocalDate date = LocalDate.parse(splits[3], formatter);
+                String dateString = synthesizeDateString(date);
+
+
+                String description = splits[4];
+
+                String parametersInput = "t/" + type + " c/" + category + " a/" + amountString + " d/" + dateString
+                        + " i/" + description;
+                command = getCommand("add", parametersInput);
+
+                ParameterParser.parse(command, parametersInput);
```

Comment:
If the user changes the file to make the format into an invalid format, for example, changing the amount in the file to "a", I believe running this line directly could throw an exception in which the error message is catered for user command execution and not a storage initialization. 

You might want to add more try catches for each possible exception and throw a more suitable error message, or to perform separate check for each individual parameter first before calling the command, for a different error message than the one used by user command execution.

Reply:
Yep. This is a good point. Currently it just throws a default file corrupted exception , cos Im still figuring a way to pin point errors at a specific line . Will update once I have the time to look into it.